contracts/core/extensions/CoreIssuanceOrder.sol

```diff
@@ -123,6 +127,51 @@ contract CoreIssuanceOrder is
         );
     }
 
+    /**
+     * Cancel an issuance order
+     *
+     * @param  _addresses      [setAddress, makerAddress, makerToken, relayerToken]
+     * @param  _values         [quantity, makerTokenAmount, expiration, relayerTokenAmount, salt]
+     * @param  _cancelQuantity Quantity of set to be filled
+     */
+    function cancelOrder(
+        address[4] _addresses,
+        uint[5] _values,
+        uint _cancelQuantity
+    )
+        external
+        isPositiveQuantity(_cancelQuantity)
+    {
+        OrderLibrary.IssuanceOrder memory order = OrderLibrary.IssuanceOrder({
+            setAddress: _addresses[0],
+            quantity: _values[0],
+            makerAddress: _addresses[1],
+            makerToken: _addresses[2],
+            makerTokenAmount: _values[1],
+            expiration: _values[2],
+            relayerToken: _addresses[3],
+            relayerTokenAmount: _values[3],
+            salt: _values[4],
+            orderHash: OrderLibrary.generateOrderHash(
+                _addresses,
+                _values
+            )
+        });
+
+        // Make sure cancel order comes from maker
+        require(order.makerAddress == msg.sender);
```

Comment:
add a message here. require(..., message)

Reply:
also, who is the maker here and who is the sender?

Reply:
Well they should be the same person. You don't want any random person to be able to cancel an order, only the order maker should be able to do that.